### PR TITLE
ci user privileges and fixes

### PR DESCRIPTION
**Pull request by mukrishn**
# Description

CI fixes, mainly to run the playbook inside a container as a non-root user. 

Fixes # (issue)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml [hunk 1]
```diff
@@ -122,6 +122,7 @@
   copy:
     dest: "/tmp/metadata.txt"
     content: "{{ lookup('template', 'node_inv.j2') | to_nice_json  }}"
+  become: true
```
Comment (smalleni) [inline ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:125]: Why is this needed?

File: ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml [hunk 2]
```diff
@@ -13,3 +13,4 @@
     echo '{{ ansible_ssh_pass }}' | sshpass ssh-copy-id -i {{ ansible_ssh_key }} -o 'StrictHostKeyChecking no' -f {{ ssh_user }}@{{ hostname }}
   changed_when: false
   when: ssh_result.stderr_lines|length > 0
+  become: true
```
Comment (smalleni) [inline ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml:16]: Why is this needed, if the orchestration host is not using the `root` user, we want whatever user is on the orchestration host to copy its keys right not the root's keys?

File: ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml (hunk 1 above)
Comment (mukrishn) [reply ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:125]: Basically `airflow` user runs a cli command which triggers the [script](https://github.com/mukrishn/airflow-kubernetes/blob/latest_patch/dags/baremetal_openshift/scripts/install_cluster.sh) inside the container.
Airflow user doesn't seems to write anything inside container without privilege escalation, `/tmp` is an `EmptyDir` volume mount. 

ex - Airflow command looks like this - `airflow tasks run 4.7_baremetal_jetski install "2021-05-11T14:55:51.829520+00:00" --local --pool default_pool --subdir /opt/airflow/dags/repo/dags/baremetal_openshift/dag.py`

File: ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml (hunk 2 above)
Comment (mukrishn) [reply ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml:16]: `sshpass` command is copying the public key from the cloned path `redhat-performance/perf-dept.git`, again airflow user couldn't ssh-copy this file so this extra privilege.

Comment (smalleni) [reply ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml:16]: OK should we debug why airflow user is not able to do this instead of fixing it here?

Comment (mukrishn) [reply ansible-ipi-install/roles/bootstrap/tasks/25_copykeys.yml]: As discussed this task copies `ansible_ssh_key` specified in `group_vars/all.yaml`. In CI, we set this to use keys from [perf-dept](https://github.com/redhat-performance/perf-dept). But anyway I tried a CI build without setting to `true` and it went thru, I am assuming the mentioned error was due to some other cause. I will revert this for now.

File: ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml (hunk 1 above)
Comment (smalleni) [reply ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:125]: so still need this?

Comment (mukrishn) [reply ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:125]: yes, this one is needed.

File: ansible-ipi-install/ci/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,59 @@
+FROM quay.io/centos/centos:8
```
Comment (mukrishn) [inline ansible-ipi-install/ci/Dockerfile:1]: @smalleni We can keep CI related files in this repo itself and I don't think we need a separate repo for this, we can delete `J-Fleet` repository or use it for some other purpose. any thoughts ? 
We don't need to read any variables from the `env_var` in airflow, all [JSON](https://github.com/mukrishn/airflow-kubernetes/blob/latest_patch/dags/openshift_nightlies/releases/4.7/baremetal/jetski/install.json) inputs are passed to the playbook as `extra_vars` and only few bash script vars are set as `env_var` like keys/tokens.

Comment (smalleni) [reply ansible-ipi-install/ci/Dockerfile:1]: sounds good.

**Review by smalleni — APPROVED** [review]
